webapp/src/components/sidebar/sidebarCategory.tsx

```diff
@@ -183,6 +184,10 @@ const SidebarCategory = (props: Props) => {
         // Capture the category ID before deletion
         const deletedFromCategoryID = props.categoryBoards.id
         
+        // Clear localStorage entries for the deleted board
+        UserSettings.setLastBoardID(teamID, null)
```

Comment:
We should only do this if the boards being deleted is the current board. Currently if you are on Board A and Delete Board B, this clears the last board ID while in fact the last last board ID should continue pointing to Board A.

Additionally, I think this is causing one more issue where if you have Board A, B, C and D in that order, and you are on D and delete B, you end up navigating to C, but you should continue to stay on D

https://github.com/user-attachments/assets/5069b68f-4bfa-4b6a-b2ff-7dbc17591833